packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,568 @@
+    @required this.keys,
+  })  : _textDirection = textDirection,
+        _selectedValue = selectedValue,
+        _pressedValue = pressedValue {
+    addAll(children);
+  }
+
+  T get selectedValue => _selectedValue;
+  T _selectedValue;
+  set selectedValue(T value) {
+    if (_selectedValue == value) {
+      return;
+    }
+    _selectedValue = value;
+    markNeedsPaint();
+  }
+
+  T get pressedValue => _pressedValue;
+  T _pressedValue;
+  set pressedValue(T value) {
+    if (_pressedValue == value) {
+      return;
+    }
+    _pressedValue = value;
+    markNeedsPaint();
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection == value) {
+      return;
+    }
+    _textDirection = value;
+    markNeedsLayout();
+  }
+
+  List<T> keys;
+
+  List<RRect> get rects => _rects;
+  final List<RRect> _rects = <RRect>[];
+
+  List<Paint> get fillPaint => _fillPaint;
+  final List<Paint> _fillPaint = <Paint>[];
+
+  static Size _boxSize(RenderBox box) => box == null ? Size.zero : box.size;
+
+  static Rect _boxRect(RenderBox box) {
+    final _SegmentedControlContainerBoxParentData childData = box.parentData;
+    return box == null ? Rect.zero : childData.offset & box.size;
+  }
+
+  static double _minWidth(RenderBox box, double height) {
+    return box == null ? 0.0 : box.getMinIntrinsicWidth(height);
+  }
+
+  static double _minHeight(RenderBox box, double width) {
+    return box == null ? 0.0 : box.getMinIntrinsicHeight(width);
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    double total = 0.0;
+    RenderBox child = firstChild;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      total += _minWidth(child, height);
+      child = childParentData.nextSibling;
+    }
+    return total;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) => computeMinIntrinsicWidth(height);
+
+  @override
+  double computeMinIntrinsicHeight(double width) => _minHeight(firstChild, width);
+
+  @override
+  double computeMaxIntrinsicHeight(double width) => computeMinIntrinsicHeight(width);
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    return firstChild.getDistanceToActualBaseline(baseline);
+  }
+
+  @override
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! _SegmentedControlContainerBoxParentData) {
+      child.parentData = new _SegmentedControlContainerBoxParentData();
+    }
+  }
+
+  @override
+  void performLayout() {
+    double maxHeight = _kMinSegmentedControlHeight;
+    final Constraints parentConstraints = constraints.loosen();
+
+    int length = 0;
+    RenderBox child = firstChild;
+
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      child.layout(parentConstraints, parentUsesSize: true);
```

Comment:
you pass in different width constraints here than lower down, so the height you get out of this doesn't necessarily match the height it will want to be. For example, say the child is `new Text('a a a a a')`. The height you'll get here is the height of one "a", but once you lay it out, it'll try to be five lines high.